app/lib/server/index.js

```diff
@@ -25,74 +25,118 @@ const getApisBlockWithUUIDs = apis => {
  * Server initialisation
  */
 const Server = function () {
-  const ssl = new SSL()
-    .useDir(config.get('server.ssl.dir'), true)
+  this.ssl = new SSL()
+    .useDomains(config.get('server.ssl.domains'))
+    .certificateDir(config.get('server.ssl.dir'), true)
+    .useEnvironment('production')
+    .provider('letsencrypt')
+    .registerTo(config.get('server.ssl.email'))
+    .autoRenew(true)
+    .byteLength(3072)
+}
 
-  this.options = {
-    port: config.get('server.ssl.enabled') ? ssl.getPort() : config.get('server.port'),
-    host: config.get('server.host'),
-    key: ssl.getKey(config.get('server.ssl.dir')),
-    certificate: ssl.getCertificate(config.get('server.ssl.dir')),
-    formatters: {
-      'text/plain': (req, res, body) => {
-        if (body instanceof Error) {
-          // catch our Error
-          log.error(body.stack)
-          return body.message
-        }
+Server.prototype.getOptions = function (override={}) {
+  const formatters = {
+    'text/plain': (req, res, body) => {
+      if (body instanceof Error) {
+        // catch our Error
+        log.error(body.stack)
+        return body.message
       }
     }
   }
+
+  return Object.assign({
+    port: config.get('server.port'),
+    host: config.get('server.host'),
+    formatters
+  }, override)
 }
 
 /**
  * Start server
  * @return {Promise}  Add Listener
  */
 Server.prototype.start = function () {
-  return new Promise((resolve, reject) => {
+  let listenerQueue = []
+  return new Promise(resolve => {
```

Comment:
I don't think we need to create a new Promise here, you could just return the result of `Promise.all`, no?